include/deal.II/matrix_free/operators.h

```diff
@@ -205,6 +397,322 @@ namespace MatrixFreeOperators
       }
   }
 
+  //----------------- Base operator -----------------------------
+  template <int dim, typename number>
+  Base<dim,number>::~Base ()
+  {
+  }
+
+
+
+  template <int dim, typename number>
+  Base<dim,number>::Base ()
+    :
+    Subscriptor(),
+    data(NULL),
+    have_interface_matrices(false)
+  {
+  }
+
+
+
+  template <int dim, typename number>
+  typename Base<dim,number>::size_type
+  Base<dim,number>::m () const
+  {
+    Assert(data != NULL,
+           ExcNotInitialized());
+    return data->get_vector_partitioner()->size();
+  }
+
+
+
+  template <int dim, typename number>
+  typename Base<dim,number>::size_type
+  Base<dim,number>::n () const
+  {
+    return m();
+  }
+
+
+
+  template <int dim, typename number>
+  void
+  Base<dim,number>::clear ()
+  {
+    data = NULL;
```

Comment:
SmartPointer does not manage any memory per default as its main purpose is to keep track of the uses of the object pointed to. So calling `clear()` is clearly wrong. However, assigning a new pointer is exactly right, as it calls `unsubscribe` on the old owner and subscribes to the new one in case it is not null. So I'd say it looks perfectly fine here.